db/inmemory_test.go

```diff
@@ -0,0 +1,54 @@
+package db
+
+import "testing"
+
+func TestBasic(t *testing.T) {
+	im := NewInMemoryVolumeDatabase()
+
+	err := im.Connect()
+	if err != nil {
+		t.Fatal("Connect returned an error! " + err.Error())
+	}
+
+	list, err := im.List()
+	if err != nil {
+		t.Fatal("List returned an error! " + err.Error())
```

Comment:
You're right, but the code after these two checks would still not run. To help me explain, here's an example of how this could be handled:

```go
list, err := im.List()
if err != nil {
	t.Error("List returned an error! " + err.Error())
}
else if len(list) != 0 {
	t.Error("List should have returned exactly zero results.")
}
```